snap/info_snap_yaml.go

```diff
@@ -21,18 +21,200 @@ package snap
 
 import (
 	"fmt"
+	"strings"
+
 	"gopkg.in/yaml.v2"
 )
 
+type snapYaml struct {
+	Name        string                 `yaml:"name"`
+	Developer   string                 `yaml:"developer"`
+	Version     string                 `yaml:"version"`
+	Type        Type                   `yaml:"type"`
+	Channel     string                 `yaml:"channel"`
+	Description string                 `yaml:"description"`
+	RawPlugs    map[string]interface{} `yaml:"plugs,omitempty"`
```

Comment:
I'd call these Plugs, Slots, and Apps.. They're not Raw.. they're just the yaml version of those, implied by the type name.